File: src/main/java/com/intetics/lukyanenko/dao/GoodsItemDAO.java
package com.intetics.lukyanenko.dao;

import com.intetics.lukyanenko.models.GoodsItem;
import javafx.util.Pair;

import java.util.ArrayList;
import java.util.List;

public interface GoodsItemDAO
        extends CommonDAO<GoodsItem>
{
  List<GoodsItem> selectWhere(ArrayList<Pair<String, Pair<String, Object>>> params);
}

File: src/main/java/com/intetics/lukyanenko/dao/GoodsItemDAO.kt
package com.intetics.lukyanenko.dao

import com.intetics.lukyanenko.models.GoodsItem
import javafx.util.Pair

import java.util.ArrayList

interface GoodsItemDAO : CommonDAO<GoodsItem> {
    fun selectWhere(params: ArrayList<Pair<String, Pair<String, Any>>>): List<GoodsItem>
}
